sklearn/neighbors/_binary_tree.pxi

```diff
@@ -230,12 +230,14 @@ leaf_size : positive int, default=40
     satisfy ``leaf_size <= n_points <= 2 * leaf_size``, except in
     the case that ``n_samples < leaf_size``.
 
-metric : str or DistanceMetric object
-    The distance metric to use for the tree.  Default='minkowski'
-    with p=2 (that is, a euclidean metric). See the documentation
-    of the DistanceMetric class for a list of available metrics.
-    {binary_tree}.valid_metrics gives a list of the metrics which
-    are valid for {BinaryTree}.
+metric : str or DistanceMetric object, default='minkowski'
```

Comment:
As discussed previously, I think it makes sense to hide that `DistanceMetric` is supported for consistency.
```suggestion
metric : str, default='minkowski'
```

Reply:
I find it weird to trade functionality for documentation consistency (as discussed [just above](https://github.com/scikit-learn/scikit-learn/pull/23806#issuecomment-1183159300) with ogrisel). But as you wish! 🤓

Reply:
Ah yes, this got lost in the shuffle. I guess your original point is right, and we can still document it.